build/Build.Steps.cs

```diff
@@ -134,11 +150,26 @@ partial class Build
 
             foreach (var app in testApps)
             {
-                DotNetBuild(x => x
-                    .SetProjectFile(app)
-                    .SetConfiguration(BuildConfiguration)
-                    .SetPlatform(Platform)
-                    .SetNoRestore(true));
+                if (TestPackageVersions.Versions.TryGetValue(app.Name, out var testedPackageVersions))
+                {
+                    foreach (var testedPackageVersion in testedPackageVersions)
+                    {
+                        DotNetBuild(x => x
+                            .SetProjectFile(app)
+                            .SetConfiguration(BuildConfiguration)
+                            .SetProperty("TestedPackageVersion", testedPackageVersion)
```

Comment:
Same suggestion of trying to use `.CombineWith` to avoid the branching.

Reply:
ba965160f94c1aad4e45de4f84e3987069e0d399